crates/carbon_app/src/managers/metadata/cache/curseforge/modpack.rs

```diff
@@ -0,0 +1,223 @@
+    app.prisma_client
+        .curse_forge_modpack_image_cache()
+        .find_unique(db::curse_forge_modpack_image_cache::project_id_file_id(
+            curseforge.project_id as i32,
+            curseforge.file_id as i32,
+        ))
+        .exec()
+        .await?
+        .ok_or_else(|| anyhow::anyhow!("No icon found for modpack"))?
+        .data
+        .ok_or_else(|| anyhow::anyhow!("No icon found for modpack"))
+}
+
+pub async fn get_modpack_metadata(
+    app: &App,
+    curseforge: CurseforgeModpack,
+) -> anyhow::Result<InstanceModpackInfo> {
+    let cache_entry = app
+        .prisma_client
+        .curse_forge_modpack_cache()
+        .find_unique(db::curse_forge_modpack_cache::project_id_file_id(
+            curseforge.project_id as i32,
+            curseforge.file_id as i32,
+        ))
+        .with(db::curse_forge_modpack_cache::logo_image::fetch())
+        .exec()
+        .await?;
+
+    let logo = cache_entry
+        .as_ref()
+        .and_then(|cache_entry| cache_entry.logo_image.as_ref())
+        .and_then(|logo_image| logo_image.as_ref().map(|logo_image| logo_image));
+
+    let is_entry_up_to_date = cache_entry
+        .as_ref()
+        .map(|entry| {
+            entry.updated_at.timestamp() + 60 * 60 * 24 * 7 > chrono::Utc::now().timestamp()
+        })
+        .unwrap_or(false);
+
+    let has_cache_entry = cache_entry.is_some();
+    let has_cache_logo = logo.is_some();
+
+    if has_cache_entry && is_entry_up_to_date {
+        let Some(cache_entry) = cache_entry else {
+            unreachable!("We just checked that cache_entry.is_some()");
+        };
+
+        return Ok(InstanceModpackInfo {
+            name: cache_entry.modpack_name,
+            version_name: cache_entry.version_name,
+            url_slug: cache_entry.url_slug,
+            has_image: cache_entry
+                .logo_image
+                .flatten()
+                .map(|logo| logo.data.is_some())
+                .unwrap_or(false),
+        });
+    } else {
+        let app = app.clone();
+        let runner = tokio::spawn(async move {
+            let modplatform_manager = app.modplatforms_manager();
+            let addon_file = modplatform_manager
+                .curseforge
+                .get_mod_file(ModFileParameters {
+                    mod_id: curseforge.project_id as i32,
+                    file_id: curseforge.file_id as i32,
+                });
+            let addon = modplatform_manager.curseforge.get_mod(ModParameters {
+                mod_id: curseforge.project_id as i32,
+            });
+
+            let (addon_file, addon) = tokio::try_join!(addon_file, addon)?;
+
+            let name = addon.data.name.clone();
+            let file_name = addon_file.data.file_name.clone();
+            let slug = addon.data.slug.clone();
+            let url = addon.data.logo.as_ref().map(|logo| logo.url.clone());
```

Comment:
You should check if the previously cached url is the same and not redownload the image if it is.